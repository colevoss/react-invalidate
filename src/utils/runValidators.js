// @flow

import type { FieldValidtor, FieldValue } from '../shared/types';

const runValidators = async (validators: Array<FieldValidtor>, value: FieldValue): Promise<string | boolean> =>
  Promise.all(validators.map(validator => validator(value)));

export default runValidators;
