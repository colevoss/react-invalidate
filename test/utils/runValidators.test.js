import runValidators from '../../src/utils/runValidators';

describe('#runValidators', () => {
  test('calls each validator', () => {
    const validators = [jest.fn().mockReturnValue(true), jest.fn().mockReturnValue(true)];

    const value = 'test';

    runValidators(validators, value);

    expect(validators).toMatchSnapshot();
  });

  test('returns true for each validator if all tests pass', async () => {
    const validators = [jest.fn().mockReturnValue(true), jest.fn().mockReturnValue(true)];

    const result = await runValidators(validators);

    expect(result).toEqual([true, true]);
  });

  test('returns failed message if one validator fails', async () => {
    const validators = [jest.fn().mockReturnValue(true), jest.fn().mockReturnValue(Promise.reject('fail'))];

    try {
      await runValidators(validators);
    } catch (message) {
      expect(message).toBe('fail');
    }
  });
});
