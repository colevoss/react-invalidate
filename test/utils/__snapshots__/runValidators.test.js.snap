// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#runValidators calls each validator 1`] = `
Array [
  [Function],
  [Function],
]
`;
